python/pylibvw.cc

```diff
@@ -655,39 +711,67 @@ void ex_push_dictionary(example_ptr ec, vw_ptr vw, py::dict& dict)
     if (ns_e().length() < 1) continue;
     py::extract<py::list> list_e(objectVal);
     py::list list = list_e();
-    char ns = ns_e()[0];
+    std::string ns_str = ns_e();
```

Comment:
ns_e() is already a string copy, you don't need to copy it into an additional buffer you can pass .c_str() directly to ex_ensure_namespace_exists

Reply:
Changed